packages/preferences/README.md

```diff
@@ -0,0 +1,154 @@
+# Preferences
+
+Utilities for storing WordPress preferences.
+
+## Installation
+
+Install the module
+
+```bash
+npm install @wordpress/preferences --save
+```
+
+_This package assumes that your code will run in an **ES2015+** environment. If you're using an environment that has limited or no support for such language features and APIs, you should include [the polyfill shipped in `@wordpress/babel-preset-default`](https://github.com/WordPress/gutenberg/tree/HEAD/packages/babel-preset-default#polyfill) in your code._
+
+## API Usage
+
+### Features
+
+Features are boolean values used for toggling specific editor features on or off.
+
+Set the default values for any features on editor initialization:
+
+```js
+import { dispatch } from '@wordpress/data';
+import { store as preferencesStore } from '@wordpress/preferences';
+
+function initialize() {
+	// ...
+
+	dispatch( preferencesStore ).setFeatureDefaults(
+		'namespace/editor-or-plugin-name',
+		{
+			myFeatureName: true,
+		}
+	);
+
+	// ...
+}
+```
+
+Use the `toggleFeature` action and the `isFeatureActive` selector to toggle features within your app:
+
+```js
+wp.data
+	.select( 'core/preferences' )
+	.isFeatureActive( 'namespace/editor-or-plugin-name', 'myFeatureName' ); // true
+wp.data
+	.dispatch( 'core/preferences' )
+	.toggleFeature( 'namespace/editor-or-plugin-name', 'myFeatureName' );
+wp.data
+	.select( 'core/preferences' )
+	.isFeatureActive( 'namespace/editor-or-plugin-name', 'myFeatureName' ); // false
+```
+
+The `MoreMenuDropdown` and `MoreMenuFeatureToggle` components help to implement an editor menu for changing preferences and feature values.
+
+```jsx
+function MyEditorMenu() {
+	return (
+		<MoreMenuDropdown>
+			{ () => (
+				<MenuGroup label={ __( 'Features' ) }>
+					<MoreMenuFeatureToggle
+						scope="namespace/editor-or-plugin-name"
+						feature="myFeatureName"
+						label={ __( 'My feature' ) }
+						info={ __( 'A really awesome feature' ) }
+						messageActivated={ __( 'My feature activated' ) }
+						messageDeactivated={ __( 'My feature deactivated' ) }
+					/>
+				</MenuGroup>
+			) }
+		</MoreMenuDropdown>
+	);
+}
+```
+
+## API Reference
```

Comment:
Should we take this opportunity to simplify further. What I have in mind could be something like:

 - drop the "feature" name (as it's a bit weird) and just call them preferences.
 - Basically the main selectors could be just `get` and `set`

Reply:
Agreed, the 'feature' thing doesn't really fit for a generic preferences package. I'll work on the naming. 👍 

I think it might good to have a `toggle` too, most of the preferences are booleans, so it seems like a handy function to have.

On this subject, I've been wondering if we want any built-in type safety for these preferences. Or whether it should just be a freeform key/value store. What do you think?

Reply:
> On this subject, I've been wondering if we want any built-in type safety for these preferences. Or whether it should just be a freeform key/value store. What do you think?

If we go the type safety road, it almost feels like we should "register" preferences. I'm not sure it's worth the added complexity. I understand that if we don't do it from the start it might harder to do later. Any use-cases that benefits from the type safety?

Reply:
> Any use-cases that benefits from the type safety?

There definitely aren't many in the editor, so I think you're right that it's not worth the complexity. The only one really is the Code Editor/Visual Editor option.

There is also the option of building more complex behavior around the existing actions and selectors. For example, if an editor needed a preference with some kind of validation, it could add that validation in its own action that calls through to `dispatch( 'core/preferences' ).set`.